src/rust/e2e/src/utils.rs

```diff
@@ -0,0 +1,145 @@
+use std::{str::FromStr, sync::atomic::AtomicU64, time::Duration};
+
+use anyhow::Result;
+use aptos_sdk::{
+    bcs,
+    move_types::{
+        ident_str,
+        language_storage::{ModuleId, StructTag, TypeTag},
+    },
+    rest_client::{aptos_api_types::MoveModuleId, FaucetClient},
+    types::{account_address::AccountAddress, transaction::EntryFunction, LocalAccount},
+};
+use clap::Parser;
+use econia_sdk::EconiaClient;
+use sqlx::PgPool;
+
+pub const TIMEOUT: Duration = Duration::from_secs(1);
+
+#[derive(Parser, Debug)]
+pub struct Args {
+    /// The URL of the Aptos node
+    pub node_url: String,
+
+    /// The URL of the faucet
+    pub faucet_url: String,
+
+    /// The address of the Econia contract
+    pub econia_address: String,
+
+    /// The address of the Aptos faucet
+    pub faucet_address: String,
+
+    /// The database URL
+    pub db_url: String,
+}
+
+pub struct State {
+    pub e_apt: TypeTag,
+    pub e_usdc: TypeTag,
+    pub faucet_address: AccountAddress,
+    pub faucet_client: FaucetClient,
+    pub econia_address: AccountAddress,
+    pub econia_client: EconiaClient,
+    pub db_pool: PgPool,
+    pub node_url: String,
+    pub market_size: AtomicU64,
+}
+
+/// Creates the initial variables needed
+pub async fn init(args: &Args) -> State {
+    // Create eAPT and eUSDC `TypeTag`s
+    let faucet_address = if args.faucet_address.starts_with("0x") {
+        args.faucet_address.clone()
+    } else {
+        format!("0x{}", args.faucet_address)
+    };
+    let faucet_address =
+        AccountAddress::from_hex_literal(&faucet_address).expect("Could not parse faucet address.");
+    let e_apt = TypeTag::Struct(Box::new(
+        StructTag::from_str(&format!("0x{faucet_address}::example_apt::ExampleAPT")).unwrap(),
```

Comment:
It either already starts with 0x, or gets prefixed with 0x, so you can get rid of the 0x prefix here I think